codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Headers.java

```diff
@@ -113,7 +120,194 @@ public static boolean isPseudoHeader(CharSequence header) {
+                    }
+                    if (header == SCHEME.value()) {
+                        return SCHEME;
+                    }
+                    if (header == STATUS.value()) {
+                        return STATUS;
+                    }
+                    // :method, :scheme, :status
+                    return getMethodSchemeStatusIfEqualsNoPrefix(array, offset);
+                case 9:
+                    if (header == PROTOCOL.value()) {
+                        return PROTOCOL;
+                    }
+                    // :protocol
+                    return getProtocolIfEqualsNoPrefix(array, offset);
+                case 10:
+                    if (header == AUTHORITY.value()) {
+                        return AUTHORITY;
+                    }
+                    // :authority
+                    return getAuthorityIfEqualsNoPrefix(array, offset);
+                }
+            }
+            return null;
+        }
+        static final int METHOD_STRING_HASHCODE = -1141949029;
+        static final int SCHEME_STRING_HASHCODE = -972381601;
+        static final int STATUS_STRING_HASHCODE = -956875604;
+        static final int PATH_STRING_HASHCODE = 56997727;
+        static final int AUTHORITY_STRING_HASHCODE = -1332238263;
+        static final int PROTOCOL_STRING_HASHCODE = 1552659666;
+        private static final boolean PSEUDO_STRING_HASHCODES_VALIDATED;
+
+        static {
+            boolean trustHashCode = true;
+            // hashCode is trust if the subsequent checks work as expected
+            trustHashCode = trustHashCode && METHOD.value().toString().hashCode() == METHOD_STRING_HASHCODE;
+            trustHashCode = trustHashCode && SCHEME.value().toString().hashCode() == SCHEME_STRING_HASHCODE;
+            trustHashCode = trustHashCode && STATUS.value().toString().hashCode() == STATUS_STRING_HASHCODE;
+            trustHashCode = trustHashCode && PATH.value().toString().hashCode() == PATH_STRING_HASHCODE;
+            trustHashCode = trustHashCode && AUTHORITY.value().toString().hashCode() == AUTHORITY_STRING_HASHCODE;
+            trustHashCode = trustHashCode && PROTOCOL.value().toString().hashCode() == PROTOCOL_STRING_HASHCODE;
+            PSEUDO_STRING_HASHCODES_VALIDATED = trustHashCode;
+        }
+
+        /**
+         * Returns the {@link PseudoHeaderName} corresponding to the specified header name.
+         *
+         * @return corresponding {@link PseudoHeaderName} if any, {@code null} otherwise.
+         */
+        public static PseudoHeaderName getPseudoHeader(String header) {
+            if (!PSEUDO_STRING_HASHCODES_VALIDATED) {
+                return getPseudoHeaderName(header);
+            }
+            // String::hashCode is intrinsified
+            // the cases uses String's literal which are the same used in the enums
+            // String::hashCode is a contract which is not going to change
+            switch (header.hashCode()) {
```

Comment:
How much benefit is this path over the `getPseudoHeaderName(header)` call above? Having symmetry with the `AsciiString` version above would make this easier to follow and cut out a lot of extract code.

Reply:
I can re-run the experiment but i can tell you already which benefits it have:
1. given that are separate methods (without reusing the existing code for AsciiString) there's no bimorphism involved around `CharSequence`, which usually mess-up with GraalVM native image, for example
2. the string's hash-code have no collisions among each others, while using the length of string(s), yep

In term of performance, I didn't designed the benchmark to pollute the type profiling of the these methods, but it is easy to do, so I can prove what happen with OpenJDK C2 as well, if you think is relevant.

The `getMethodSchemeStatusIfEqualsNoPrefix` is what happen in  case of worst case matches, which is not great, if compared to what we can have now with `String`s and distinct hashcodes.

Reply:
Sorry, I wasn't clear. My suggestion was only regarding using the hashcode instead of switching on length. The benchmark of a String without a populated hashcode wasn't tested, although that starts to come back to the point you made earlier that we're going to be computing a hashcode anyway. That kind of suggests that the benchmarking should be done at the level of adding a header to the header map: this change might only matter in that context.
I'm just coming from a position that we have a lot of code here that has the same end goal but takes different approaches which is odd to me. The switch on length seems "good enough" and should work for all cases.